src/sentry/integrations/discord/views/link_identity.py

```diff
@@ -27,41 +23,44 @@ def build_linking_url(integration: RpcIntegration, discord_id: str) -> str:
     return absolute_uri(reverse(endpoint, kwargs={"signed_params": sign(salt=SALT, **kwargs)}))
 
 
-@control_silo_view
-class DiscordLinkIdentityView(BaseView):
-    """
-    Django view for linking user to Discord account.
-    """
-
-    @method_decorator(never_cache)
-    def handle(self, request: HttpRequest, signed_params: str) -> HttpResponse:
-        try:
-            params = unsign(signed_params, salt=SALT)
-        except (SignatureExpired, BadSignature):
-            return render_to_response("sentry/integrations/discord/expired-link.html")
-
-        organization, integration, idp = get_identity_or_404(
-            ExternalProviders.DISCORD,
-            request.user,
-            integration_id=params["integration_id"],
-        )
-
-        if request.method != "POST":
-            return render_to_response(
-                "sentry/auth-link-identity.html",
-                request=request,
-                context={"organization": organization, "provider": integration.get_provider()},
-            )
-
-        Identity.objects.link_identity(user=request.user, idp=idp, external_id=params["discord_id"])  # type: ignore[arg-type]
-
-        analytics.record(
-            "integrations.discord.identity_linked",
-            provider="discord",
-            actor_id=request.user.id,
-            actor_type=ActorType.USER,
-        )
-        return render_to_response(
-            "sentry/integrations/discord/linked.html",
-            request=request,
-        )
+class DiscordLinkingView(LinkingView, ABC):
```

Comment:
Can you move this to its own file since this one is called link identity?

Reply:
Sure. I made that change in https://github.com/getsentry/sentry/pull/76214 and I've pulled it down into this branch.

I'm also adopting the term "linkage" in class names, to try to encompass both linking and unlinking.